packages/storage-ui/src/Components/Modules/LoginModule/PasswordlessEmail.tsx

```diff
@@ -0,0 +1,299 @@
+    },
+    buttonLink: {
+      color: palette.additional["gray"][10],
+      outline: "none",
+      textDecoration: "underline",
+      cursor: "pointer",
+      textAlign: "center",
+      marginTop: constants.generalUnit * 2,
+      "&.disabled": {
+        color: palette.additional["gray"][7]
+      },
+      "&.spaceLeft": {
+        marginLeft: constants.generalUnit * 0.5
+      }
+    },
+    title: {
+      textAlign: "center"
+    },
+    subtitle: {
+      textAlign: "center",
+      paddingTop: constants.generalUnit * 4,
+      paddingBottom: constants.generalUnit * 4
+    },
+    resendText: {
+      textAlign: "center",
+      marginTop: constants.generalUnit * 4
+    },
+    errorText: {
+      textAlign: "center",
+      color: palette.error.main
+    }
+  })
+)
+
+interface IPasswordlessEmail {
+  resetLogin: () => void
+}
+
+const PasswordlessEmail = ({ resetLogin }: IPasswordlessEmail) => {
+  const classes = useStyles()
+  const [isSubmitEmailLoading, setIsSubmitEmailLoading] = useState(false)
+  const [isSubmitResendEmailLoading, setIsSubmitResendEmailLoading] = useState(false)
+  const [isSubmitNonceLoading, setIsSubmitNonceLoading] = useState(false)
+  const [page, setPage] = useState<"confirmEmail" | "confirmVerificationCode">("confirmEmail")
+  const [email, setEmail] = useState<string | undefined>()
+  const { storageApiClient, login } = useStorageApi()
+  const [hasEmailResent, setHasEmailResent] = useState(false)
+  const [error, setError] = useState<string | undefined>()
+
+  const emailValidation = useMemo(() => yup.object().shape({
+    email: yup
+      .string()
+      .email("Please enter a valid email")
+      .required(t`Email is required`)
+  })
+  , [])
+
+  const nonceValidation = useMemo(() => yup.object().shape({
+    nonce: yup
+      .string()
+      .required(t`Verification code is required`)
+  })
+  , [])
+
+  const onSubmitEmail = useCallback((values) => {
+    setIsSubmitEmailLoading(true)
+    setError(undefined)
+    storageApiClient.getIdentityEmailToken({ email: values.email })
+      .then(() => {
+        setEmail(values.email)
+        setPage("confirmVerificationCode")
+        setIsSubmitEmailLoading(false)
+      }).catch ((e) => {
+        setError(t`Something went wrong!`)
+        setIsSubmitEmailLoading(false)
+        console.error(e)
+      })
+  }, [storageApiClient])
+
+  const onSubmitNonce = useCallback((values) => {
+    if (!email) return
+    setIsSubmitNonceLoading(true)
+    setError(undefined)
+    storageApiClient.postIdentityEmailToken({
+      email: email,
+      nonce: values.nonce
+    }).then(async (data) => {
+      await login("email", { token: data || "", email })
+      setIsSubmitNonceLoading(false)
+    }).catch ((e) => {
+      if (e && e[0] && e[0].type === "nonce") {
+        setError(t`Verification code not correct!`)
+      } else {
+        setError(t`Something went wrong!`)
+      }
+      setIsSubmitNonceLoading(false)
+      console.error(e)
+    })
```

Comment:
using finally here too.
```suggestion
    }).then(async (data) => {
      await login("email", { token: data || "", email })
    }).catch ((e) => {
      if (e && e[0] && e[0].type === "nonce") {
        setError(t`Verification code not correct!`)
      } else {
        setError(t`Something went wrong!`)
      }
      console.error(e)
    }).finally(() => setIsSubmitNonceLoading(false))
```